standardization/src/test/scala/za/co/absa/enceladus/standardization/interpreter/StandardizationInterpreter_DecimalSuite.scala

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.standardization.interpreter
+
+import java.text.{DecimalFormat, NumberFormat}
+import java.util.Locale
+import org.apache.spark.sql.types.{DecimalType, StringType, StructField, StructType}
+import org.scalatest.FunSuite
+import za.co.absa.enceladus.utils.error.{ErrorMessage, UDFLibrary}
+import za.co.absa.enceladus.utils.testUtils.SparkTestBase
+
+class StandardizationInterpreter_DecimalSuite extends FunSuite with SparkTestBase {
+  import spark.implicits._
+
+  private implicit val udfLib: UDFLibrary = new UDFLibrary
+
+  private val desiredSchema = StructType(Seq(
+    StructField("description", StringType, nullable = false),
+    StructField("small", DecimalType(5,2), nullable = false),
+    StructField("big", DecimalType(38,18), nullable = true)
+  ))
+
+  private val zero = BigDecimal("0E-18")
+  private val bigDecimalFormat = {
+    val pattern = "0.000000000000000000" //18 decimal places
+    val nf = NumberFormat.getNumberInstance(Locale.US)
+    val df = nf.asInstanceOf[DecimalFormat]
+    df.applyPattern(pattern)
+    df
+  }
+
+  private def bd(number: Double): BigDecimal = {
+    val s: String =  bigDecimalFormat.format(number)
+    BigDecimal(s)
+  }
+
+  test("From String") {
+    val seq = Seq(
+      ("01-Pi", "3.14", "3.14"),
+      ("02-Null", null, null),
+      ("03-Long", Long.MaxValue.toString, Long.MinValue.toString),
+      ("04-infinity", "-Infinity", "Infinity"),
+      ("05-Really big", "123456789123456791245678912324789123456789123456789.12",
+        "12345678912345679124567891232478912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912345678912346789123456789123456789123456789123456791245678912324789123456789123456789123456789123456789123456791245678912324789123456789123456789123456789123456789123456789123456789123456789.1"),
```

Comment:
Could you use concatenation and cut the string?

Reply:
I could. What benefit do you see in that?
The point is, that it's something really big value, the actual value doesn't really matter to be seen, IMO.

Reply:
I am moving through the file using arrows with the cursor being at the end of the line or I am moving the lines. I need to scroll to check if brackets and commas are correct and what is at the end.